src/Features/Core/Portable/EmbeddedLanguages/RegularExpressions/RegexParser.cs

```diff
@@ -259,18 +262,30 @@ private RegexExpressionNode ParseAlternatingSequences(bool consumeCloseParen)
         /// 
         /// An empty sequence just means "match at every position in the test string".
         /// </summary>
-        private RegexExpressionNode ParseAlternatingSequencesWorker(bool consumeCloseParen)
+        private RegexAlternationNode ParseAlternatingSequencesWorker(
+            bool consumeCloseParen, bool isConditional)
         {
-            RegexExpressionNode current = ParseSequence(consumeCloseParen);
+            using var _ = ArrayBuilder<RegexNodeOrToken>.GetInstance(out var builder);
+            builder.Add(ParseSequence(consumeCloseParen));
 
             while (_currentToken.Kind == RegexKind.BarToken)
             {
                 // Trivia allowed between the | and the next token.
-                current = new RegexAlternationNode(
-                    current, ConsumeCurrentToken(allowTrivia: true), ParseSequence(consumeCloseParen));
+                var barToken = ConsumeCurrentToken(allowTrivia: true);
+                if (isConditional && builder.Count > 1)
```

Comment:
Took me a while to parse this in my head. I think `>= 3` is easier to read and understand, meaning we already saw `Sequence|AnotherSequence` and we don't need more than.

Both should be functionally equivalent, so it's a minor suggestion one can agree or disagree with.

Reply:
sure.  i don't mind changing.